e2e/test/AppContextTests.cs

```diff
@@ -105,28 +151,59 @@ public async Task AppConfig_ReceiveAsyncAfterCloseAsync_x509_AmqpWs()
 
             if (isFlagSet)
             {
-                await DCLC_ReceiveAsyncAfterCloseAsync(TestDeviceType.X509, Client.TransportType.Amqp_WebSocket_Only).ConfigureAwait(false);
+                await ReceiveAsyncAfterCloseAsync(TestDeviceType.X509, Client.TransportType.Amqp_WebSocket_Only, false).ConfigureAwait(false);
+            }
+            else
+            {
+                Assert.Fail("AppContext flag DisableObjectDisposedExceptionForReceiveAsync not set");
+            }
+        }
+
+        [TestMethod]
+        public async Task AppConfig_ReceiveAsyncWithCancellationTokenAfterCloseAsync_x509_AmqpWs()
+        {
+            bool isFlagSet = false;
+            bool flag = false;
+
+#if !NET451
+            isFlagSet = AppContext.TryGetSwitch(AppContextConstants.DisableObjectDisposedExceptionForReceiveAsync, out flag);
```

Comment:
Isn't this failing on NET451? Did you run `dotnet test` within `e2e\tests` and it passed?